keyboards/cipulot/common/ec_switch_matrix.h

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2023 Cipulot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "matrix.h"
+#include "eeconfig.h"
+
+typedef struct {
+    uint8_t  actuation_mode;                              // 0: normal board-wide APC, 1: Rapid trigger from specific board-wide actuation point, 2: Rapid trigger from resting point
+    uint16_t mode_0_actuation_threshold;                  // threshold for key press in mode 0
+    uint16_t mode_0_release_threshold;                    // threshold for key release in mode 0
+    uint16_t mode_1_initial_deadzone_offset;              // threshold for key press in mode 1
+    uint8_t  mode_1_actuation_sensitivity;                // sensitivity for key press in mode 1 and 2 (1-255)
+    uint8_t  mode_1_release_sensitivity;                  // sensitivity for key release in mode 1 and 2 (1-255)
+    uint16_t bottoming_reading[MATRIX_ROWS][MATRIX_COLS]; // bottoming reading
+} eeprom_ec_config_t;
```

Comment:
May want to add `PACKED` (also from `util.h`) to ensure the structs only take the least space posible.

Reply:
Tested and confirmed working.